### DISCUSSION TITLE
Intermittent failures -- Ubuntu desktop on MacOS host

### DISCUSSION THREAD

**Discussion by zipzit** [ruzickap/packer-templates#379] [Q&A]
I've been a big fan of peru/ubuntu-XXXX-desktop-amd64 Vagrant box (and different Ubuntu versions over the past few years.). I use that box as the base of my Vagrantfile, running in VirtualBox and add a few things in.  In the latest update, I'm getting really odd intermittent failures.  The box may work once in awhile, and othertimes just hang.  

I submitted my general issue to the VirtualBox Forum for  "VirtualBox on Mac OS X Hosts" .  I got some feedback.  Some I understand, and some that just isn't clear.  The one issue with that forum is the absolutely do not support Vagrant.  Seems an odd decision to me, but I can see how that gets out of hand.  

I'm still struggling to understand, REALLY UNDERSTAND,  how to troubleshoot a VirtualBox hangup.  Yup, there is a log file for the VirtualBox but lots of entries don't seem connected to the failure.  All of a sudden, this box is a fail on MacOS.  It may run once in a while, but more often it just hangs.  

The help from the VirtualBox forum was to reduce the allocation for CPU cores and memory.  I've used those values for years, but who knows perhaps there is something to that.. Easy fix...

the other issue I'm not really clear about is the Guest Additions installation.  

From the VirtualBox forum: 

_Next error is that you have "foreign" Guest Additions installed. These are not controlled by the Oracle devs and so are not guaranteed to work, nor in fact are they supported at all.
00:00:15.374009 VMMDev: Guest Additions information report: Version 6.1.26 r145957 '6.1.26_Ubuntu'

The version number of the genuine article does not include that "_Ubuntu" modifier. Install the genuine GAs bundled with the software._

Not sure whats going on here.  Where does Guest Additions Version 6.1.26 r145957 '6.1.26_Ubuntu' come from?  I'm assuming that is built into the peru/ubuntu-20.04-desktop-amd64 .  Is that a valid assumption?  

Note, I'm running:
--VirtualBox 6.1.34
--MacOS Monterey Version 12.3.1
--Vagrant 2.2.19

Any ideas on why a box that used to work great now fails?  Many thanks.... 

My Vagrantfile is attached.
[Vagrantfile.zip](https://github.com/ruzickap/packer-templates/files/8588288/Vagrantfile.zip)

**Comment by ruzickap** [comment]
Hello.

I'm using VirtualBox Guest Additions which are part of the Ubuntu distribution:

* https://github.com/ruzickap/packer-templates/blob/main/scripts/ubuntu/virtualbox-guest-x11.sh
* https://github.com/ruzickap/packer-templates/blob/main/scripts/ubuntu/virtualbox-guest.sh

Troubleshooting hanging virtual machine in VirtualBox may be really complex and difficult...
